nowcasting_dataloader/datasets.py

```diff
@@ -179,6 +187,183 @@ def __getitem__(self, batch_idx: int) -> dict:
+        src_path: str,
+        tmp_path: str,
+        configuration: Configuration,
+        cloud: str = "gcp",
+        required_keys: Union[Tuple[str], List[str]] = None,
+        history_minutes: Optional[int] = None,
+        forecast_minutes: Optional[int] = None,
+        normalize: bool = True,
+        add_position_encoding: bool = False,
+        add_satellite_target: bool = False,
+        add_hrv_satellite_target: bool = False,
+    ):
+        """
+        Netcdf Dataset
+
+        Args:
+            n_batches: Number of batches available on disk.
+            src_path: The full path (including 'gs://') to the data on
+                Google Cloud storage.
+            tmp_path: The full path to the local temporary directory
+                (on a local filesystem).
+            cloud:
+            required_keys: Tuple or list of keys required in the example for
+                it to be considered usable
+            history_minutes: How many past minutes of data to use, if subsetting the batch
+            forecast_minutes: How many future minutes of data to use, if reducing the amount
+                of forecast time
+            configuration: configuration object
+            cloud: which cloud is used, can be "gcp", "aws" or "local".
+            normalize: normalize the batch data
+            add_position_encoding: Whether to add position encoding or not
+            add_satellite_target: Whether to add future satellite imagery to the target or not
+            add_hrv_satellite_target: Whether to add future HRV satellite imagery to the target
+        """
+        super().__init__(
+            n_batches=n_batches,
+            src_path=src_path,
+            tmp_path=tmp_path,
+            configuration=configuration,
+            cloud=cloud,
+            required_keys=required_keys,
+            history_minutes=history_minutes,
+            forecast_minutes=forecast_minutes,
+            normalize=normalize,
+            add_position_encoding=add_position_encoding,
+        )
+
+        self.add_satellite_target = add_satellite_target
+        self.add_hrv_satellite_target = add_hrv_satellite_target
+
+        # SatFlow specific changes, i.e. which timestep to split on
+        self.current_timestep_index = (self.history_minutes // 5) + 1
+        self.current_timestep_index_30 = (self.history_minutes // 30) + 1
+
+    def __getitem__(self, batch_idx: int) -> Tuple[dict, dict]:
+        """
+        Satflow extension for the dataloader
+
+        Args:
+            batch_idx: Batch ID to load
+
+        Returns:
+            Tuple of dicts of torch.Tensors holding the data, with the first one being the inputs
+            and the second being the target
+        """
+        batch = super().__getitem__(batch_idx)
+        x = {}
+        target = {}
+        # Need to partition out past and future sat images here, along with the rest of the data
+        if len(batch["satellite"].get("data", [])) > 0:
```

Comment:
I wonder if you want to move it back to a `Batch` class, or any option to return it like that. 

That'll make things like 'batch["nwp"]["data"]' go to batch.nwp.data whcih is think is better. 

Could put this perhaps as a TODO

Reply:
Yeah, this would involve putting the position encodings into the `Batch` or `BatchML` as well, so probably put it as a TODO for now?